proposals/0836-lean-core-jsc.md

```diff
@@ -0,0 +1,33 @@
+---
+title: Extract JSC (JavaScriptCore) from React Native Core
+author:
+- Kudo Chien <kudo@expo.dev>
+date: 2024-11-28
+---
+
+# RFC0836: Extract JSC (JavaScriptCore) from React Native Core
+
+## Summary
+
+This RFC proposes a Lean Core effort to extract JSC (JavaScriptCore) from React Native's core.
+
+## Proposed Changes and Timeline
+
+- Version 0.77: Remove the [`jsc-android` npm dependency](https://github.com/facebook/react-native/blob/2d337efc23c662143b3f39a6c994d80fec047054/packages/react-native/package.json#L132) from React Native and instead download the artifact from Maven Central. When using Hermes, downloading the unused npm dependency is unnecessary and removing it saves about 32 MB of uncompressed size.
```

Comment:
Is this needed in 0.77 for 16KB page support?

Reply:
yes, but if there's concern to cherry-pick https://github.com/facebook/react-native/pull/47972 to 0.77. we can postpone the mavenCentral migration to 0.78. on 0.77 we still use jsc-android npm package for a newer version with android 16kb page support.

Reply:
I'd rather go with 0.78.

Our take on 16K support is that it's Hermes first. Originally we were not even planning to include JSC for 16K support.